v1/writable/image_test.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writable_test
+
+import (
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-containerregistry/v1"
+	"github.com/google/go-containerregistry/v1/tarball"
+	"github.com/google/go-containerregistry/v1/writable"
+)
+
+// If we don't perform any operations the
+// writable image should be identical to it's source
+func TestSourceEquality(t *testing.T) {
+	source := sourceImage(t)
+
+	image, err := writable.NewImage(source)
+	if err != nil {
+		t.Fatalf("Unexpected error creating a writable image: %v", err)
+	}
+
+	if !manifestsAreEqual(t, source, image) {
+		t.Fatal("manifests are not the same")
```

Comment:
This can be `t.Errorf` since it might be helpful to see whether the configs matched even if manifests didn't.